pkgs/applications/office/imag/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv
+, lib
+, rustPlatform
+, fetchFromGitHub
+, llvmPackages
+, openssl
+, pkg-config
+, Security
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "imag";
+  version = "0.10.0";
+
+  src = fetchFromGitHub {
+    owner = "matthiasbeyer";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1rkl4n3x2dk9x7705zsqjlxh92w7k6jkc27zqf18pwfl3fzz8f8p";
+  };
+
+  buildInputs = [ openssl pkg-config ]
+      ++ lib.optional stdenv.isDarwin Security;
+
+  LIBCLANG_PATH = "${llvmPackages.libclang}/lib";
+
+  cargoPatches = [ ./cargo-lock.patch ];
```

Comment:
@matthewbauer I understand the noise concern, and yet, consider:

- https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
- https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

> If you’re building an end product, which are executable like command-line tool or an application, or a system library with crate-type of `staticlib` or `cdylib`, check Cargo.lock into `git`.

Currently, because lock file is not checked in, it's impossible to replicate the exact set of dependencies that you're developing `imag` with.

Reply:
You mean me, not Matthew. Yes, I consider checking it in for the next release in the release branch (not in the master branch though).

A patch release is pending (not sure though when I want to release it), I could add the Cargo.lock to the patchset release!

Reply:
So sorry for the typo!